lib/js_of_ocaml/js.mli

```diff
@@ -813,6 +827,24 @@ external to_bytestring : js_string t -> string = "caml_string_of_jsbytes"
       Javascript string should only contain UTF-16 code points below
       255.) *)
 
+external float : float -> float = "caml_js_from_float"
```

Comment:
The signature looks wrong. Why don't we have `float -> number t` ?

Reply:
This is so that our users can write code that works with this version of the library, where we are using `float` throughout for JavaScript numbers, and a version of the library where the type of JavaScript numbers is `number t`.

By the way, I'm using `float t` for now, since I was concerned of a possible performance impact of using an object type. I need to check...

Reply:
I'd be easier to read if we had an alias.

Also, if I understand correctly, this is just a temporary solution if we end up merging jsoo and wasm_of_ocaml.
We'll eventually need this library to no use float directly. Is that correct ?

Reply:
Yes, this is only temporary, to make the switch smoother. 

I can add a `number_t` type (or do you have a better name?) and use it throughout the Js_of_ocaml library, if you like.

Reply:
> I can add a number_t type (or do you have a better name?) and use it throughout the Js_of_ocaml library, if you like.

Sounds good, it makes the code a bit clearer and should also help with the transition.

Reply:
It seems that examples have not been updated in the PRs.
Can you also check the manual ? Or do you prefer to leave it for later ?

Reply:
The main objective of this PR is to undeprecate the float conversion functions and to add missing primitives.
I need to check the manual at some point, indeed. But maybe that can wait.

Reply:
Ok. The diff looks good. I'll let you rebase and merge